index.js

```diff
@@ -180,4 +180,21 @@ function importNS(ns) {
     return module;
 }
 
-module.exports = importNS('EosKnowledgeContent');
+const Eknc = importNS('EosKnowledgeContent');
+
+// Setup async functions
+Eknc.Engine.prototype.get_object_for_app = function (id, app_id) {
+    return new Promise((resolve, reject) => {
+        bindings.EngineGetObject(this, id, app_id, resolve, reject);
```

Comment:
Not sure how this is working, because the `resolve` and `reject` args aren't being read any longer in `EngineGetObject()`. And IIUC you are just getting a promise returned directly from there, so I don't think you have to wrap it in another promise here.